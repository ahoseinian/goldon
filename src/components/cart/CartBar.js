import React from 'react'
import CartIcon from './CartIcon'
// import SearchButton from '../search/SearchButton'
import HomeLink from '../common/Links/HomeLink'

const CartBar = ({pathname}) => (
    <div className="d-flex justify-content-between">
      {pathname !== '/' && <HomeLink/>}
      <CartIcon/>
    </div>
)


CartBar.propTypes = {
  pathname: React.PropTypes.string.isRequired
}

export default CartBar
