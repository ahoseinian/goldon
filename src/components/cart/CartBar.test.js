import React from 'react'
import {shallow} from 'enzyme'
import CartBar from './CartBar'
import toJson from 'enzyme-to-json'

describe('CartBar', () => {
  it('renders', () => {
    const wrapper = shallow(<CartBar pathname="/"/>)
    expect(
      console.error.called
    ).toEqual(false)
  })

  it('does not render HomeLink on home', () => {
    const wrapper = shallow(<CartBar pathname="/" />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('does renders HomeLink on othe pathes', () => {
    const wrapper = shallow(<CartBar pathname="/gg" />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
