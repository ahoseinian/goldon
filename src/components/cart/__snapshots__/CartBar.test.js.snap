exports[`CartBar does not render HomeLink on home 1`] = `
<div
  className="d-flex justify-content-between">
  <Connect(Cart) />
</div>
`;

exports[`CartBar does renders HomeLink on othe pathes 1`] = `
<div
  className="d-flex justify-content-between">
  <Component />
  <Connect(Cart) />
</div>
`;
